python/cuspatial/cuspatial/geometry/geoarrowbuffers.py

```diff
@@ -0,0 +1,640 @@
+        cuspatial operations. If `pandas`, GeoArrowBuffer is stored on Host and
+        used for creating GeoPandas objects.
+        """
+        return self._data_location
+
+    @data_location.setter
+    def data_location(self, data_location):
+        if data_location not in [cudf, pd]:
+            raise NotImplementedError(
+                "only cudf and pandas CoordinateArrays "
+                "are supported at this time"
+            )
+        else:
+            self._data_location = data_location
+
+    def _serialize(self, data):
+        try:
+            if self._data_location == pd:
+                if isinstance(data, cudf.Series):
+                    return data.to_pandas()
+                else:
+                    return self._data_location.Series(data)
+            elif self._data_location == cudf:
+                return self._data_location.Series(data)
+            else:
+                raise ValueError(
+                    f"{type(self._data_location)} incorrect data"
+                    f"location: pandas or cudf"
+                )
+        except Exception as e:
+            raise TypeError(e, f"{type(data)} is not supported in cuspatial")
+
+    @property
+    def xy(self):
+        """
+        The coordinates of this Geometry Column in interleaved format
+        [x,y,...,x,y].
+        """
+        return self._xy
+
+    @xy.setter
+    def xy(self, xy):
+        if (len(xy) % 2) != 0:
+            raise ValueError("xy must have even length")
+        self._xy = self._serialize(xy)
+
+    @property
+    def z(self):
+        """
+        An optional third dimension for this Geometry.
+        """
+        return self._z
+
+    @z.setter
+    def z(self, z):
+        self._z = self._serialize(z)
+
+    def __getitem__(self, index):
+        if isinstance(index, slice):
+            new_index = slice(index.start * 2, index.stop * 2 + 1, index.step)
+            return self.xy.iloc[new_index]
+        return self.xy.iloc[(index * 2) : (index * 2) + 2]
+
+    def __repr__(self):
+        return f"xy:\n" f"{self.xy.__repr__()}\n"
+
+    def copy(self, deep=True):
+        """
+        Create a copy of all points.
+
+        Parameters
+        ----------
+        deep : Boolean.
+            If set to False, a new object referencing the same
+            set of GPU memory will be created. If True, GPU memory will be
+            copied.
+        """
+        if self.z is not None:
+            z = self.z.copy(deep)
+        else:
+            z = None
+        result = type(self)(self.xy.copy(deep), z)
+        return result
+
+    @property
+    def x(self):
+        """
+        Return packed x-coordinates of this GeometryArray object.
+        """
+        return self.xy[slice(0, None, 2)].reset_index(drop=True)
+
+    @property
+    def y(self):
+        """
+        Return packed y-coordinates of this GeometryArray object.
+        """
+        return self.xy[slice(1, None, 2)].reset_index(drop=True)
```

Comment:
+1 to question about `@cached_property` here as well

Reply:
`@cached_property` is only in python 3.8, but we're still supporting 3.7 and 3.8, so I reverted it back to @propery until 3.7 is no longer supported.